plugins/flytekit-dgxc-lepton/flytekitplugins/dgxc_lepton/connector.py

```diff
@@ -0,0 +1,749 @@
+import asyncio
+import json
+import logging
+import re
+import subprocess
+import sys
+import tempfile
+import time
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Any, Dict, Optional
+
+from flyteidl.admin.agent_pb2 import GetTaskLogsResponse
+from flyteidl.core.execution_pb2 import TaskExecution, TaskLog
+
+from flytekit import current_context
+from flytekit.extend.backend.base_connector import AsyncConnectorBase, ConnectorRegistry, Resource, ResourceMeta
+from flytekit.extend.backend.utils import convert_to_flyte_phase
+from flytekit.loggers import logger
+from flytekit.models.literals import LiteralMap
+from flytekit.models.task import TaskTemplate
+
+
+# Configure logging to ensure our messages go to stdout
+def setup_logging():
+    """Setup logging to ensure our connector logs are visible - respecting ROOT_LOG_LEVEL env var"""
+    import os
+
+    # Get log level from environment variable, default to DEBUG
+    root_log_level = os.environ.get("ROOT_LOG_LEVEL", "DEBUG").upper()
+    log_level_map = {
+        "DEBUG": logging.DEBUG,
+        "INFO": logging.INFO,
+        "WARNING": logging.WARNING,
+        "ERROR": logging.ERROR,
+        "CRITICAL": logging.CRITICAL,
+    }
+    root_level = log_level_map.get(root_log_level, logging.DEBUG)
+
+    root_logger = logging.getLogger()
+    # Only add handler if none exists to avoid duplicates
+    if not any(isinstance(h, logging.StreamHandler) and h.stream == sys.stdout for h in root_logger.handlers):
+        handler = logging.StreamHandler(sys.stdout)
+        handler.setFormatter(logging.Formatter("%(asctime)s - %(name)s - %(levelname)s - %(message)s"))
+        root_logger.addHandler(handler)
+
+    # Set root logger level based on environment variable
+    root_logger.setLevel(root_level)
+    print(f"ROOT LOGGER LEVEL SET TO: {root_log_level} ({root_level})")
+
+    # Also configure the flytekit logger specifically for DEBUG
+    flytekit_logger = logging.getLogger("flytekit")
+    flytekit_logger.setLevel(logging.DEBUG)
+
+    # Configure our connector logger for DEBUG
+    connector_logger = logging.getLogger(__name__)
+    connector_logger.setLevel(logging.DEBUG)
```

Comment:
should this also be "WARNING"? if everything's WARNING, we probably don't really need a separate connector logger no?